app/scripts/controllers/replicaSet.js

```diff
@@ -0,0 +1,448 @@
+            details: "Reason: " + $filter('getErrorDetails')(e)
+          };
+        });
+    };
+
+    $scope.clearEnvVarUpdates = function() {
+      copyDeploymentAndEnsureEnv($scope.deployment);
+      $scope.forms.envForm.$setPristine();
+    };
+
+    var limitWatches = $filter('isIE')() || $filter('isEdge')();
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+        // FIXME: DataService.createStream() requires a scope with a
+        // projectPromise rather than just a namespace, so we have to pass the
+        // context into the log-viewer directive.
+        $scope.projectContext = context;
+
+        var allHPA = {}, limitRanges = {};
+        var updateHPA = function() {
+          $scope.hpaForRS = HPAService.filterHPA(allHPA, kind, $routeParams.replicaSet);
+          if ($scope.isActive) {
+            // Show both HPAs that target the RC and the DC if this is the active deployment.
+            var hpaForDC = HPAService.filterHPA(allHPA, 'DeploymentConfig', $scope.deploymentConfigName);
+            $scope.autoscalers = $scope.hpaForRS.concat(hpaForDC);
+          } else {
+            $scope.autoscalers = $scope.hpaForRS;
+          }
+        };
+
+        var watchActiveDeployment = function() {
+          // Watch all replication controllers so we know if this is the active deployment to enable scaling.
+          watches.push(DataService.watch($scope.resource, context, function(deployments) {
+            var activeDeployment,
+                deploymentsForConfig = [],
+                getAnnotation = $filter("annotation");
+            // Filter the list to just those deployments for this config.
+            angular.forEach(deployments.by("metadata.name"), function(deployment) {
+              var depConfigName = getAnnotation(deployment, 'deploymentConfig') || "";
+              if (depConfigName === $scope.deploymentConfigName) {
+                deploymentsForConfig.push(deployment);
+              }
+            });
+            activeDeployment = DeploymentsService.getActiveDeployment(deploymentsForConfig);
+            $scope.isActive = activeDeployment && activeDeployment.metadata.uid === $scope.deployment.metadata.uid;
+            updateHPA();
+          }));
+        };
+
+        var pods, selector;
+        var updatePodsForDeployment = function() {
+          if (!pods || !selector) {
+            return;
+          }
+
+          $scope.podsForDeployment = _.filter(pods, function(pod) {
+            return selector.matches(pod);
+          });
+        };
+
+        var updateHPAWarnings = function() {
+            HPAService.getHPAWarnings($scope.deployment, $scope.autoscalers, limitRanges, project)
+                      .then(function(warnings) {
+              $scope.hpaWarnings = warnings;
+            });
+        };
+
+        var updateDC = function(rc) {
```

Comment:
> looked like you updated other references to rc vars to be rs, update this one too?

I left this one as `rc` because it's always a replication controller if it has a deployment config (at least for now).